lib/puppet_blacksmith/rake_tasks.rb

```diff
@@ -129,8 +129,8 @@ def define(args, &task_block)
         task :push => :build do
           m = Blacksmith::Modulefile.new
           forge = Blacksmith::Forge.new
-          puts "Uploading to Puppet Forge #{forge.username}/#{m.name}"
-          forge.push!(m.name)
+          puts "Uploading to Puppet Forge #{m.author}/#{m.name}"
+          forge.push!(m.author, m.name)
```

Comment:
Unrelated to this PR, but I'd love to fix the logic here to use the version from the Modulefile and upload that package instead of relying on the general selection we have now.

Reply:
I've added an additional optional parameter to `push!` with the version from the metadata.